sotodlib/io/load_smurf.py

```diff
@@ -662,35 +748,53 @@ def add_new_observation(
             The active session
         calibration : boolean
             Boolean that indicates whether the observation is a calibration observation.
-        max_early : int
+        session_id : int (optional, but much more efficient)
+            session id, if known, for timestream files that should go with the observations
+        max_early : int (optional)
             Buffer time to allow the g3 file to be earlier than the smurf action
         """
+                
+        if session_id is None:
+            x = session.query(Files.name)
+            x = x.filter(
+                Files.stream_id == stream_id,
+                Files.start >= dt.datetime.utcfromtimestamp(action_ctime - max_early)
+            )
+            x = x.order_by(Files.start).first()
+            if x is None:
+                logger.debug(
+                    f"No .g3 files from Action {action_name} in {stream_id}"
+                    f" at {action_ctime}. Not Making Observation"
+                )
+                return
+
+            session_id = int((x.name[:-3].split("/")[-1]).split("_")[0])
+        
+        if calibration:
+            obs_id=f"oper_{stream_id}_{session_id}"
+        else:
+            obs_id=f"obs_{stream_id}_{session_id}"
```

Comment:
Noticing the `Observations` docstrings aren't up to date and don't have the `obs/oper` prefix.